splitio/storage/inmemmory.py

```diff
@@ -107,6 +107,236 @@ def remove_flag_set(self, flag_sets, feature_flag_name, should_filter):
                 if self.flag_set_exist(flag_set) and len(self.get_flag_set(flag_set)) == 0 and not should_filter:
                     self._remove_flag_set(flag_set)
 
+class InMemoryRuleBasedSegmentStorage(RuleBasedSegmentsStorage):
+    """InMemory implementation of a feature flag storage base."""    
+    def __init__(self):
+        """Constructor."""
+        self._lock = threading.RLock()
+        self._rule_based_segments = {}
+        self._change_number = -1
+
+    def get(self, segment_name):
+        """
+        Retrieve a rule based segment.
+
+        :param segment_name: Name of the segment to fetch.
+        :type segment_name: str
+
+        :rtype: splitio.models.rule_based_segments.RuleBasedSegment
+        """
+        with self._lock:
+            return self._rule_based_segments.get(segment_name)
+
+    def update(self, to_add, to_delete, new_change_number):
+        """
+        Update rule based segment.
+
+        :param to_add: List of rule based segment. to add
+        :type to_add: list[splitio.models.rule_based_segments.RuleBasedSegment]
+        :param to_delete: List of rule based segment. to delete
+        :type to_delete: list[splitio.models.rule_based_segments.RuleBasedSegment]
+        :param new_change_number: New change number.
+        :type new_change_number: int
+        """
+        [self._put(add_segment) for add_segment in to_add]
+        [self._remove(delete_segment) for delete_segment in to_delete]
+        self._set_change_number(new_change_number)
+
+    def _put(self, rule_based_segment):
```

Comment:
Should it be double underscored if it is a private method (`__`)?

Reply:
nope, in python its one underscore, the double underscore are operators overload functions